### ISSUE TITLE
linter: False positive for exhaustive-deps when using recursive useCallback

### ISSUE THREAD

**Issue by florianwittmann** [oxc-project/oxc#11433]
### What version of Oxlint are you using?

0.17.0

### What command did you run?

`oxlint -c .oxlintrc.json`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "categories": {
    "correctness": "error"
  },
  "plugins": ["react", "unicorn", "typescript", "oxc", "import", "promise"],
  "rules": {
    "no-unused-vars": [
      "warn",
      {
        "args": "none",
        "caughtErrors": "none",
        "ignoreRestSiblings": true,
        "vars": "all"
      }
    ]
  }
}

```

### What happened?

When using useCallback to define a recursive function, oxlint incorrectly reports a missing dependency for the function itself, even though it is not reported as such from react-hooks/exhaustive-deps in eslint.

```ts
  const recursive = useCallback(
    (n: number): number => (n <= 0 ? 0 : n + recursive(n - 1)),
    [],
  );
```

Expected behavior: No lint error, since _recursive_ is intentionally recursive and should not be added to its own dependency array.

Actual behavior: oxlint reports:
> React Hook useCallback has a missing dependency: 'recursive'
help: Either include it or remove the dependency array.oxc[eslint-plugin-react-hooks(exhaustive-deps)](https://oxc.rs/docs/guide/usage/linter/rules/react/exhaustive-deps.html)

**Comment by camc314** [comment]
fixed in #11438
